### PR TITLE
add bitswap stats command, close #2302

### LINKED ISSUE #2302: Bitswap Stats Command
### Description
Print bitswap statistics via a command.  This is a followup from #2067 which hacked this in inline.  To do this cleaner we instead want a command to get these stats.

### Acceptance criteria

This code should exist in a bitswap wrapper on porcelain or wherever we keep these infrequently used networking calls:
```
stat, err := blockService.Exchange().(*bitswap.Bitswap).Stat()                                                                                              
if err != nil {                                                                                                                                               
   return nil,nil,nil, err                                                                                   
}                                                                                                                                                             
return  stat.DupBlksReceived, stat.DataReceived, stat.BlocksReceived, nil
```
### Where to begin
Take a look at the DHT command and the stub code above.

### PR DESCRIPTION
**Pull request by contrun**
Add bitswap stats command and a corressponding plumbing api.
Closes #2302

### CODE REVIEW

File: commands/bitswap.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	},
+
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		stats, err := GetPorcelainAPI(env).BitswapGetStats(req.Context)
+		if err != nil {
+			return err
+		}
+		return res.Emit(stats)
+	},
+	Type: &bitswap.Stat{},
```
Comment (ZenGround0) [inline commands/bitswap.go]: NON-blocking: FYI to @anorth @mishmosh , I thought about requesting a pretty printed text encoder for this command, but piping the existing json output to `jq` makes this very readable.

File: commands/main.go [hunk 2]
```diff
@@ -107,6 +107,7 @@ VIEW DATA STRUCTURES
   go-filecoin show                   - Get human-readable representations of filecoin objects
 
 NETWORK COMMANDS
+  go-filecoin bitswap                - Explore libp2p bitswap
```
[the remark this answers is not in the document]
Comment (anorth) [reply commands/main.go]: I'm not sure, so let's keep this as-is now and revise as part of a larger CLI usability effort informed by some better research & design.

File: commands/bitswap.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package commands
+
+import (
+	"gx/ipfs/QmXixGGfd98hN2dA5YiPHWANY3sjmHfZBQk3mLiQUo6NLJ/go-bitswap"
```
[the remark this answers is not in the document]
Comment (contrun) [reply commands/bitswap.go]: Updated this with go mod!

**Comment by ZenGround0** [comment]
@contrun looks like you've introduced some issues into tests.  Not sure what's going on exactly but some tests aren't building.

**Comment by contrun** [comment]
@ZenGround0 I did notice that, but I really don't understand what's happening. The tests of my another pull request https://github.com/filecoin-project/go-filecoin/pull/2461 which obviously did not introduce any significant change also failed inexplicably. I will look into the details tomorrow.

**Comment by mishmosh** [comment]
We think this is failing CI because outside contributors' jobs get run as `Medium` resource class. @ognots has researched and found there's no great way around this. We'd need to make our tests less memory intensive (less parallelism in go tests?) or have a committer re-run tests from forked where resource class is downgraded to medium.

Ergo, @ZenGround0 as reviewer with the right permissions could you re-run tests and help shepherd to completion?

**Comment by contrun** [comment]
@mishmosh Thank you for your information. I actually tried to do some of the failed tests, they surprisingly succeeded. But as there are simply too many failed tests, I didn't reran the failed tests completely.

**Pull request merged**